Formula/opencl-icd-loader.rb

```diff
@@ -0,0 +1,31 @@
+class OpenclIcdLoader < Formula
+  desc "OpenCL Installable Client Driver (ICD) Loader"
+  homepage "https://www.khronos.org/registry/OpenCL/"
+  url "https://github.com/KhronosGroup/OpenCL-ICD-Loader/archive/refs/tags/v2022.05.18.tar.gz"
+  sha256 "71f70bba797a501b13b6b0905dc852f3fd6e264d74ce294f2df98d29914c4303"
+  license "Apache-2.0"
+  head "https://github.com/KhronosGroup/OpenCL-ICD-Loader.git", branch: "main"
+
+  keg_only :shadowed_by_macos, "macOS provides OpenCL.framework"
+
+  depends_on "cmake" => :build
+  depends_on "opencl-headers" => [:build, :test]
+
+  def install
+    inreplace "loader/icd_platform.h", "\"/etc/", "\"#{etc}/"
```

Comment:
Probably should add a caveat similar to `ocl-icd` (and add comment for `inreplace`) as our directory doesn't match official specification. Sadly, `opencl-icd-loader` is less friendly to using non-standard location and requires modifying code rather than allowing config-time option.

Reply:
I think we can just copy the caveat from `ocl-icd` as it's basically the same.

As for the `inreplace`, this is certainly not the only formula where we have to replace hard-coded prefixes so I think as long as it's explained it should be okay.